### PR TITLE
fix(Dropdown, ComboBox): adds listener to open/close menu on enter

### LINKED ISSUE #2823: AVT 2 - React Dropdown - Enter key does not expand and collapse the list
## Environment
Tested on macOS Mojave 10.14.5, Version 74.0.3729.157 (Official Build) (64-bit), with VoiceOver

## Description
Tab to the Dropdown box
Hit the Enter

The Enter key does not expand or collapse the listbox. See [ARIA Authoring Practices Guide ](https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html)for additional information. 

It states the following:

Enter key: 

-  If the focus is on the button, expands the listbox and places focus on the currently selected option in the list.
- If focus is in the listbox , collapses the listbox and keeps the currently selected option as the button label.

### PR DESCRIPTION
**Pull request by abbeyhrt**
Closes #2823 

Components using Downshift, like Dropdown and ComboBox, were not opening on enter, this adds `toggleMenu` from Downshift to toggle the menu when `isOpen` changes. 

*Note: this problem also affects MultiSelect, since it uses Downshift but is more complicated to fix. I've made an issue (#3861) and will work on a solution for that separately. 

#### Changelog

**Changed**

- add toggleMenu and onKeydown listener to `ListBox.Field` to toggleMenu on enter

#### Testing / Reviewing

Go to ComboBox and Dropdown, tab into them and make sure they toggle open/close with Enter press.

### CODE REVIEW

**Review by jnm2377 — APPROVED** [review]
LGTM! 🎉

File: packages/react/src/components/ComboBox/ComboBox.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ import React from 'react';
 import { settings } from 'carbon-components';
 import { WarningFilled16 } from '@carbon/icons-react';
 import ListBox, { PropTypes as ListBoxPropTypes } from '../ListBox';
+import { match } from '../../internal/keyboard/match';
+import * as keys from '../../internal/keyboard/keys';
```
Comment (tw15egan) [inline packages/react/src/components/ComboBox/ComboBox.js]: Can we simplify this to be `import { match, keys } from '../../internal/keyboard';`?

File: packages/react/src/components/Dropdown/Dropdown.js [hunk 2]
```diff
@@ -12,6 +12,8 @@ import React from 'react';
 import { settings } from 'carbon-components';
 import { WarningFilled16 } from '@carbon/icons-react';
 import ListBox, { PropTypes as ListBoxPropTypes } from '../ListBox';
+import { match } from '../../internal/keyboard/match';
+import * as keys from '../../internal/keyboard/keys';
```
Comment (tw15egan) [inline packages/react/src/components/Dropdown/Dropdown.js]: Same as above

**Review by tw15egan — APPROVED** [review]
Tiny style nits for consistency otherwise looks great! 👍✅

File: packages/react/src/components/ComboBox/ComboBox.js (hunk 1 above)
Comment (abbeyhrt) [reply packages/react/src/components/ComboBox/ComboBox.js]: Yeah for sure, made the change!

**Review by joshblack — APPROVED** [review]

**Pull request merged**